LanguageFeatures/Patterns/logical_and_A05_t01.dart

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion
+/// logicalAndPattern ::= ( logicalAndPattern '&' )? relationalPattern
+///
+/// A pair of patterns separated by & matches only if both subpatterns match.
+/// Unlike logical-or patterns, the variables defined in each branch must not
+/// overlap, since the logical-and pattern only matches if both branches do and
+/// the variables in both branches will be bound.
+///
+/// If the left branch does not match, the right branch is not evaluated. This
+/// only matters because patterns may invoke user-defined methods with visible
+/// side effects.
+///
+/// @description Checks that if the left branch does not match, the right branch
+/// is not evaluated
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns
+
+import "patterns_lib.dart";
+import "../../Utils/expect.dart";
+
+main() {
+  Shape shape1 = Circle(1);
+  switch (shape1) {
+    case Circle(area: 2) & Circle(area: 3.14):
+      Expect.fail("Pattern should not match");
+      break;
+    default:
+  }
+  Expect.equals("Circle.area=2;", shape1.log);
```

Comment:
We can't really see the difference between "the rightmost object pattern wasn't matched" and "the rightmost pattern was matched, but using the cached value for `area`", so we should use two distinct getters and see that the second one isn't called.

Reply:
Fixed